src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -85,5 +133,219 @@ public override bool Execute()
 
             return true;
         }
+
+        private void AddDefaultMappingsForOnlineSources(XElement pkgSrcMappingClearElement, XElement pkgSourcesElement)
+        {
+            foreach (string sourceName in pkgSourcesElement
+                .Descendants()
+                .Where(e => e.Name == "add" && !allSourcesPackages.Keys.Contains(e.Attribute("key").Value))
+                .Select(e => e.Attribute("key").Value)
+                .Distinct()
+                .ToArray())
```

Comment:
ToArray is unnecessary overhead here isn't it?

Reply:
Oops - you're right - will fix.